docs/guide/publish.md

```diff
@@ -0,0 +1,63 @@
+# Building and Publishing
+
+Rye currently uses [build](https://github.com/pypa/build) to build the package and uses [twine](https://github.com/pypa/twine) to publish it.
+
+## Build
+
+By default, `rye` will build the both sdist and wheel target in the `dist` directory.
+
+```
+rye build
+```
+
+You can using the `--sdist` or `--wheel` flag to build the specific target, or specify the output directory with `--out`.
+
+```
+rye build --wheel --out target
+```
+
+If you want to clean the build directory before building, run:
+
+```
+rye build --clean
+```
+
+## Publish
+
+Rye will publish the distribution files under the `dist` directory to PyPI by default.
+
+```bash
+rye publish
+```
+
+You maybe be asked to input your access token and some other info if needed.
+
+```
+No access token found, generate one at: https://pypi.org/manage/account/token/
+Access token:
+
+```
+
+You can also specify the distribution files to be published:
+
+```
+rye publish dist/example-0.1.0.tar.gz
+```
+
+### --repository
+
+Rye support publish the package to a different repository by using the `--repository` and `--repository-url` flags. For example, to publish to the test repository:
+
+```
+rye publish --repository testpypi --repository-url https://test.pypi.org/legacy/
+```
+
+### --yes
+
+You can optionally set the `--yes` flag to skip the confirmation prompt, this can be useful for CI/CD pipelines.
+
+```
+rye publish --token <your_token> --yes
+```
+
+In the backend, rye save the credentials info in the `$HOME/.rye/credentials`, it means you don't need to provide the credentials info next time.
```

Comment:
```suggestion
Rye will save and re-use credentials that are stored in `$HOME/.rye/credentials`.
```

Reply:
In fact, `rye` stores not only token, but also repository, username, etc. This modification may not be comprehensive enough?

Maybe:
```
Rye will store your repository info in `$HOME/.rye/credentials` for future use.
```

Reply:
Agreed!